### ISSUE TITLE
Highlighting "leaves" in GraphViz output?

### ISSUE THREAD

**Issue by jpalardy** [martido/brew-graph#6]
Hi there,

I've been using brew-graph for a while and it's quite helpful. In fact, I noticed I can't depend on the `brew leaves` output because it doesn't account for "requirements". For example: vim depends on perl, but perl is a "requirement" rather than a "dependency", so perl is considered a "leaf".

I wrote an awk script to find nodes without any dependencies -- aka leaf nodes -- from the output of `brew graph`. If a node is never on the right side of an arrow, nobody depends on it.

If I incorporated this logic directly in brew-graph, would that be something you would be willing to merge?

In practice, the `create_node` function could take a 2nd argument, a boolean for leaf. Something like:

    def create_node(node, leaf)
      %Q(  "#{node}"#{leaf ? " [style=filled]" : ""};)
    end

It opens some questions:

* is this something you could mirror for the GraphML output? (this is something I don't use)
* is this something that would require a command-line flag? or would you consider this a reasonable behavior for the default output?

**Comment by jpalardy** [comment]
![highlighted-leaves](https://cloud.githubusercontent.com/assets/18142/21511388/1f56c93e-cc55-11e6-8aae-4c1cf2b326b5.png)

A sample of how `[style=filled]` changes the output -- leaves are gray.

**Comment by martido** [comment]
Hi,

That's a nice idea. It's actually also pretty simple to do something similar for GraphML. You just specify a different fill color for the nodes. Logic should be the same as in your solution. If you don't mind, I'll extend it to include the changes for GraphML, too.

Also, I'd like to add a command line option for this to give people the choice. Just not sure if it should be enabled by default or not.

**Comment by jpalardy** [comment]
Sounds good.

I blogged about this -- http://blog.jpalardy.com/posts/untangling-your-homebrew-dependencies/ -- once you've made the improvements, I'll edit the post to only link to your repo.

**Comment by martido** [comment]
Just released a new version that includes this feature. It is enabled by adding the `--highlight-leaves` command line switch. It's disabled by default.

Btw, have you ever used the `--all` switch to create the dependency graph of all formulae? I'm asking because I'm thinking about removing it. It used to work, but I recently noticed that the resulting Dot and GraphML files are unusable. The picture that is created by dot cannot be opened in any image viewer that I tried, and the GraphML file doesn't load in yEd on my machine.
